src/internal/base-component/__tests__/metrics.test.ts

```diff
@@ -68,10 +78,15 @@ describe('Client Metrics support', () => {
 
     describe('within an iframe', () => {
       mockConsoleError();
+      const originalWindowParent = Object.getOwnPropertyDescriptor(window, 'parent')!;
+
+      afterEach(() => {
+        Object.defineProperty(window, 'parent', originalWindowParent);
+        expect(window.parent.AWSC).toBeUndefined();
+      });
+
       const setupIframe = () => {
-        const parentWindow = { ...window };
-        Object.defineProperty(window, 'parent', { value: parentWindow });
-        Object.defineProperty(parentWindow, 'parent', { value: parentWindow });
+        Object.defineProperty(window, 'parent', { configurable: true, writable: true, value: { parent: {} } });
       };
```

Comment:
There was an incorrect iframe setup which leaked its state to all other tests (`window.parent` was left overridden).